lego/apps/users/views/abakus_groups.py

```diff
@@ -43,3 +47,21 @@ def get_queryset(self):
             return AbakusGroup.objects_with_text.prefetch_related("users").all()
 
         return self.queryset
+
+    @action(detail=False, methods=["GET"])
+    def random_interests(self, request):
+        queryset = self.get_queryset().filter(type="interesse")
+
+        values = queryset.values_list("pk", flat=True)
+        if not values:
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        values = list(values)
+
+        if len(values) <= 3:
+            instance = [queryset.get(pk=pk) for pk in values]
+        else:
+            instance = [queryset.get(pk=pk) for pk in sample(values, 3)]
+
+        serializer = self.get_serializer(instance, many=True)
```

Comment:
A more performant and "database" way to do this is to programatically create a single query, and only evaluate that. Instead of here, where you create three different queries that just has a difference in the pk (in addition to the top one). 

One way to do this could be to create a list of the `pk`s you need in python and evaluate one queryset:
```python
random_pks = sample(values, 3)
random_qs = queryset.filter(pk__in=random_pks)
```

You could also use `Q`s:
```python
from django.db.models import Q
...
random_qs = [Q(pk=pk) for pk in sample(values, 3)]
instances = queryset.filter(reduce(operator.or_, random_qs))
```
This one is more just to highlight that `Q` (and `F`) is a thing that exists to build more complicated queries. In this case the first one is of course way better. But then you have seen it at least :wink:

Reply:
Ouf thanks for pointing it out^

It looks much cleaner with a single query :heart_eyes: 

> I also do agree that this should be a generic endpoint. I do understand that you don't want to introduce more logic, even though I fear that it won't happen if you don't do it now (that might just be a good thing if we don't ever have a need for it though shrug)

I feel like this is a case where i am comfortable not being conservative. Besides, adding a generic endpoint would require changing the same file, hopefully they see/reuse some of the code and consider changing it in the one function which uses it in the frontend:3

> I figured that we can generate a hash on the client, f.ex. using their username and the current day/hour/week (however long we want to keep the current values), and send this hash to the endpoint. We can then create a random list if ids, and store it in redis with this hash as a key. Then, we can just fetch the existing list of ids from redis the next time around, and fetch those objects

Didn't know this was an option:o I'll keep this solution in mind if users find the session-based randomness distracting:)